kdl_parser/include/kdl_parser/kdl_parser.hpp

```diff
@@ -61,8 +61,10 @@ bool treeFromFile(const std::string & file, KDL::Tree & tree);
  * \param tree The resulting KDL Tree
  * returns true on success, false on failure
  */
+#if defined(HAS_ROS) && defined(HAS_URDF)
```

Comment:
Mind removing `HAS_ROS` and `HAS_URDF` from the header file? If I understand correctly all downstream ROS users will need to define `HAS_ROS` and `HAS_URDF` before this file is included if they want to use this method. Instead I think`treeFromParam()` can return `false` if it was built without ROS support.